rfcs/2021-01-22-spatial-system-rfc.md

```diff
@@ -0,0 +1,159 @@
+
+# Summary
+**Spatial System** is a system with a set of rules and principles on how to measure `size` & `spacing` of your UI elements. Spacing is important, generally which is overlooked as a part of the visual design.
+
+# Motivation
+### **Why do we need a spatial system?**
+Without space none of your elements get seen, instead they become noise. There is a great relationship that exsist between space and elements. By controlling and shaping space in our designs, we create rhythm, direction, & motion.
+
+Uniformity in spacing & sizing doesn't only allows the product to look more consistent but it also helps designers and developers to communicate better & reduce number of decisions a designer has to make on daily-basis.
+
+There are so many inconsistencies which we see in the current UI, here are a few examples:
+![Inconsistent Spacing & Alignment](./images/spatial/uneven-spacing.png)
+
+### **Expected Outcome**
+Due to these inconsistencies the users using our product can get a feeling of untrustworthy, cheap design, & inconsistency. When everything looks aligned it creates a visual rhythm which in turns look pleasing and something our users expect from a brand like us especially in the fintech space.
+
+To avoid this, we need to build a flexible **spatial system** that can drive minor everyday decisions and make our designs looks consistent, organised & pleasing to our users eyes.
+
+# Detailed Design
+### **What is a pt?**
+A point (`pt`) is a measurement of space that is dependent on screen resolution. Simply put, at a `1x` resolution (or `@1x`). That is,
+```
+1pt = 1px
+```
+>**📝Note:** It is always recommended to design in **`1x`** and then exporting your assets to different sizes (`@2x`, `@3x`, etc.) depending upon the use case.
+
+### **Space ≠ Grid**
+Industry generally considers space as grids, but these two are different things. Spaces are used to make grids (i.e. grids/containers/etc. are built using spaces).
+
+### **8pt grid system**
+Before diving into the **8pt grid system** we need to consider few things where this spatial system will be used. We'll use this in overall layout for `components` (molecules, organisms, templates) as well as on the `typography`.
+
+>**📝Note:** This system will used across mobiles, tablets & desktops.
+
+### **But why 8pt grid?**
+There is no right or wrong answer here, but there are few reasons for choosing 8pt over all others (`5pt`, `6pt`, `7pt`...)
+- It scales perfectly in all type of screen displays, be it Android or iOS (`@0.75` or `@1.5`).
+- Apple & Google advice to use the same, without stating too many reasons though.
+- It's simple for us to do the math. The `4` & `8` are simpler to calculate and they are flexible enough to setup a scale.
+
+### **Base Unit**
+We'll start with the base unit, this will allow us to create a scale of all the sizes in your spatial system. For us this also means using multiples of `x8` to define dimensions, padding, and margin of elements.
```

Comment:
Maybe complete the sentence on the lines of `but x8 won't work for us let's see in the below section` because when I just read this point I got carried away with the thought that `x8` will just be too much in multiple terms and then I saw the section `What system will work for Razorpay`